### ISSUE TITLE
Honor the XDG basedir specification

### ISSUE THREAD

**Issue by piegamesde** [JoseExposito/touchegg#525]
### Is your feature request related to a problem? Please describe

Currently, the search path for potential configuration files is hard-coded.

### Describe the solution you'd like

Do an environment variable lookup as described in the [XDG base directory specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) to determine the appropriate location(s).

### Additional context

Related to https://github.com/JoseExposito/gnome-shell-extension-x11gestures/issues/21, and probably supersedes #510.

**Comment by JoseExposito** [comment]
Thanks for the bug report.

$XDG_CONFIG_DIRS looks like the variable to use. Could you copy and paste the value of that variable on NixOS, please?

I'll test with a similar value on my OS to make sure it solves the problem.

**Comment by piegamesde** [comment]
You want `$XDG_CONFIG_DIRS` for reading config, and `$XDG_CONFIG_HOME` if the application wants to edit the configuration on a user level.

```sh
$XDG_CONFIG_DIRS=/etc/xdg:/root/.nix-profile/etc/xdg:/etc/profiles/per-user/root/etc/xdg:/nix/var/nix/profiles/default/etc/xdg:/run/current-system/sw/etc/xdg
```

Note that most of these entries symlink to the same folder and are thus redundant, but this is not necessarily always the case. Furthermore, many applications are wrapped and the wrapper script sets some environment variables. Touchégg for example will probably get an entry pointing to its install location, `/nix/store/SOMEVERLYONGHASH-touchegg-VERSION/etc/xdg`.

**Comment by JoseExposito** [comment]
Cool, thanks for the example.

> Touchégg for example will probably get an entry pointing to its install location, /nix/store/SOMEVERLYONGHASH-touchegg-VERSION/etc/xdg.

Will the GNOME Shell extension and [Touché](https://github.com/JoseExposito/touche) have the same path in their $XDG_CONFIG_DIRS variable?

If not, the issues running them on NixOS will remain.

**Comment by piegamesde** [comment]
> If not, the issues running them on NixOS will remain.

Yes but they'll be a lot more manageable. This is a really common problem, so we have tooling to automatically wrap a binary while setting some environment variables. This is way less painful than having to manage a patch set.
